src/nbody/nbody_utils.hpp

```diff
@@ -75,7 +93,7 @@ static void PrintSystem(const int npart, ParArray1D<Particle> particles) {
 //----------------------------------------------------------------------------------------
 //! \fn  void NBody::SyncWithRebound
 //! \brief Copy the positions and velocities from the rebound sim to the Particles list
-static void SyncWithRebound(struct reb_simulation *r_sim, std::vector<int> particle_id,
+static void SyncWithRebound(RebSim r_sim, std::vector<int> particle_id,
```

Comment:
```suggestion
static void SyncWithRebound(RebSim &r_sim, std::vector<int> particle_id,
```
Probably should always pass this by ref

Reply:
and you might want to delete the copy constructors completely

Reply:
I think it's fine to pass this by copy (maybe there was an issue with an older version, but now that I'm holding a `shared_ptr` that doesn't care about pass by copy vs reference (although reference is slightly more performant so I did switch). 

I think it's fine to keep the copy constructor now due to the `shared_ptr`.

Reply:
Okay, sounds good. I think you need to push those changes though, I don't see it.

Reply:
Pushed -- I was worried about
```c++
RebSim r_sim;
  if (parthenon::Globals::my_rank == 0) {
    // Advance the simulation.  If this is the final stage, advance the master simulation.
    // If not, advance a copy of the master.
    int sid = disable_stderr();
    r_sim = (stage < nstages) ? RebSim(reb_sim) : reb_sim;
    ```
    but I think this is actually fine because for all nodes we later require an initialized `r_sim` when doing the MPI sync